File: retos-python/04_decorando_arbol.py
# No se imprime la base en el navegador
def create_xmas_tree(height, ornament):
    HASHTAG = '#' 
    tree = []

    for i in range(height):
        line = '_' * (height - i - 1)
        orna = ornament * ((2 * i) + 1)
        tree.append(line + orna + line)

    
    base = '_' * (height - 1)
    
    tree.append(base + HASHTAG + base)
    tree.append(base + HASHTAG + base)
    
    return '\n'.join(tree)

File: retos-ts/04_decorando_arbol.ts
// Resultado 5 ⭐
function createXmasTree(height: number, ornament: string): string {
    let tree: string = '';
    for (let i = 0; i < height; i++) {
        let line = '_'.repeat(height - i - 1);
        let orna = ornament.repeat((2 * i) + 1);

        tree += line + orna + line + '\n';
    }

    const base: string = '_'.repeat(height - 1);

    tree += base + '#' + base + '\n';
    tree += base + '#' + base;

    return tree;
}